### ISSUE TITLE
Doesn't show all trove games

### ISSUE THREAD

**Issue by Bertaz** [UncleGoogle/galaxy-integration-humblebundle#70]
**Describe the bug**
Some games from the trove don't show up in the integration. Not sure if caused by  gogcom/galaxy-integrations-python-api#72, because I can't find the Game ID.
Right now there are 98 games in the trove, the integration shows only 87 so 11 are missing, all of the newly added: ![recently added](https://user-images.githubusercontent.com/44445647/70848338-533e8c80-1e70-11ea-9cb7-5c40a0568c26.png) plus some others like Operator.

Steps to reproduce: Edit config.ini to only show trove games > Sync Integration

**Expected behavior**
All the trove games should be added by the integration and show up in gog galaxy.

**Plugin installed**
Directly from Galaxy

**Provide logs**
[plugin-humble-logs.zip](https://github.com/UncleGoogle/galaxy-integration-humblebundle/files/3963825/plugin-humble-logs.zip)

**Comment by Bertaz** [comment]
I analyzed the log a little bit, there are currently 98 games in the trove:
![troveFind](https://user-images.githubusercontent.com/44445647/70850894-7a588680-1e8f-11ea-9171-39fc4505c527.png)
and the integration finds 94, so it's actually skipping the recently added.

I found the GameIDs, 5 of the 7 not showing up are `"type": "spam"` so it's a galaxy problem, but the remaining 2 are `"type": "dlc"`.
They are `avadon3_hintbookandbonuses_trove` dlc of `avadon3_thewarborn_trove`
& `starwars_xwingvstiefighter_trove` dlc of itself. Ideally they should show up in the game they are dlc for.

**Comment by UncleGoogle** [comment]
I'm getting trove games from 
https://www.humblebundle.com/api/v1/trove/chunk?index=0
https://www.humblebundle.com/api/v1/trove/chunk?index=1
https://www.humblebundle.com/api/v1/trove/chunk?index=2
https://www.humblebundle.com/api/v1/trove/chunk?index=3
https://www.humblebundle.com/api/v1/trove/chunk?index=4
Currently there is 94 in total, those reported by plugin. Could you help me find which are thsoe 4 lacking titles? Are they present on https://www.humblebundle.com/subscription/trove ?

> starwars_xwingvstiefighter_trove dlc of itself.

What do you mean?

**Comment by Bertaz** [comment]
> Currently there is 94 in total, those reported by plugin. Could you help me find which are thsoe 4 lacking titles? Are they present on <https://www.humblebundle.com/subscription/trove>?
> 
> EDIT Are they those 4 recently added?

Exactly, you can see them in the image in the opening comment. The one with the strange font is called Arrog.
As those api calls don't report them I guess this is another gog bug. Would you mind reporting it on mantis2.gog.com? As I guess you would be more detailed in the report than me.
> > starwars_xwingvstiefighter_trove dlc of itself.
> 
> What do you mean?

If you go to [gamesdb.gog/starwars_xwingvstiefighter](https://gamesdb.gog.com/platforms/humble/external_releases/starwars_xwingvstiefighter_trove) you can see that `starwars_xwingvstiefighter_trove` (it is the base game + the dlc) has type: dlc with parent id: null. I'm not sure if it's gog's fault, either way I reported it on mantis2 with the spam games.

**Comment by UncleGoogle** [comment]
> As those api calls don't report them I guess this is a HB bug. Would you mind reporting it to HB? 

I afraid its not a bug, but humble page get those information from somewhere else. Additionally I see that  https://www.humblebundle.com/subscription/trove do not ask for 
https://www.humblebundle.com/api/v1/trove/chunk?index=0  (the newest games lies there)
Only chunks 1-4.
![545a493703c4c3876ddd2a10bc1af77e](https://user-images.githubusercontent.com/12197825/70867628-c0891500-1f77-11ea-8474-b4b19cf095c5.png)

Maybe I need to research the javascript there... Any help appreciated

**Comment by ayush-j** [comment]
I think HB has fixed this on the API side as the newly added games are now available in chunk 1. The only problem left now is games being marked as spam.

> I afraid its not a bug, but humble page get those information from somewhere else. Additionally I see that https://www.humblebundle.com/subscription/trove do not ask for
> https://www.humblebundle.com/api/v1/trove/chunk?index=0 (the newest games lies there)
> Only chunks 1-4.

The data from chunk 0 (plus some additional metadata about newly added games) is already embedded in the HTML page inside a script tag labelled "webpack-monthly-trove-data". That's probably why it doesn't request chunk 0.

**Comment by UncleGoogle** [comment]
Thanks for help @ayush-j, I've added PR.
The info in json you've found contains also:
```
 "countdownTimerOptions": {
      "currentTime": "2019-12-30T20:48:37.105434",
      "nextAdditionTime": "2020-01-10T18:00:00"
    },
```
Let us see if at 2020-01-10 the problem will come back for fresh trove content. I so, I'll merge.

PS there are also 2 things to investigate:
- check if trove cache if working properly - when coded I've wrongly assumed that new trove games will come in increased "chunk" numeration. But it is not true
- use "standardProducts" from scrapped json instead of chunk/index=0 - it they do so on web, it is more safe to follow this way

**Comment by UncleGoogle** [comment]
New January trove games: "Unarallel" and "Flames in the flood" are shown only in "recent" games on the webpage.

Merging changes - will be released on 0.6.0 soon :)

Thanks for contributing!
